modules/fakturabank/model/fakturabank.class.php

```diff
@@ -1875,16 +1875,14 @@ public function hash_to_xml($InvoiceO) {
                 }
 
                 if (!empty($InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID)){
-                    if (strlen(preg_replace('/[^0-9]/', '', $InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID)) == 9) { // has valid org nr, add it,
-                        $legal_entity = $doc->createElement('cac:PartyLegalEntity');
-                        $cbc = $doc->createElement('cbc:CompanyID', utf8_encode($InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID));
-                        $cbc->setAttribute('schemeID', 'NO:ORGNR');
-                        $legal_entity->appendChild($cbc);
+                  $legal_entity = $doc->createElement('cac:PartyLegalEntity');
+                  $cbc = $doc->createElement('cbc:CompanyID', utf8_encode($InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID));
+                  $cbc->setAttribute('schemeID', utf8_encode($InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyIDSchemeID));
+                  $legal_entity->appendChild($cbc);
 
-                        $cacparty->appendChild($legal_entity);
-                    } else {
-                        $_lib['message']->add("hash_to_xml::invalid orgnr. Organisasjonsnummeret " . $InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID . " til " . $InvoiceO->AccountingCustomerParty->Party->PartyName->Name . " er ugyldig. Fakturaen ble likevel sendt med kundenr som id.");
-                    }
+                  $cacparty->appendChild($legal_entity);
+                } else {
+                    $_lib['message']->add("hash_to_xml::invalid firma id. Firma ID " . $InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID . " til " . $InvoiceO->AccountingCustomerParty->Party->PartyName->Name . " er ugyldig. Fakturaen ble likevel sendt med kundenr som id.");
```

Comment:
Can you check if we are validating `$InvoiceO->AccountingCustomerParty->Party->PartyLegalEntity->CompanyID` anywhere earlier. If we already validate it, remove the else block and the conditional.

Change the message to "Firma ID til " . $InvoiceO->AccountingCustomerParty->Party->PartyName->Name . " er ugyldig. Fakturaen ble ikke sendt"
and return so we don't send.